### PR TITLE
Queries added

### PR DESCRIPTION
**Pull request by DavidH3031**
Could you please let me know if the way i have done orderLookup on line 20 of news.model.js is okay.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -47,6 +47,80 @@ describe("/api/articles", () => {
...
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            comment_count: expect.any(String),
+          });
+        });
+      });
+  });
+  it("GET - 200: Should return an array in ascending order.", () => {
+    return request(app)
+      .get("/api/articles?order=oldest")
```
Comment (Geo-Beard) [inline __tests__/app.test.js]: This should be either asc or desc - the ticket specifies this - oldest makes sense but it's not what the ticket asks for.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -47,6 +47,80 @@ describe("/api/articles", () => {
...
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            comment_count: expect.any(String),
+          });
+        });
+      });
+  });
+  it("GET - 200: Should return an array in ascending order sorted by article ID with the topic of cats.", () => {
+    return request(app)
+      .get("/api/articles?order=oldest&sort_by=article_id&topic=cats")
```
Comment (Geo-Beard) [inline __tests__/app.test.js]: Similar to above - asc or desc as per the ticket. If you're working as a developer and you change what the ticket asks for to "oldest" and this gets passed to a front end team it will cause confusion later down the line.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -175,6 +249,42 @@ describe("Error Handling", () => {
...
+        });
+    });
+    it("GET- 400: Should respond with 400 when all but one query is valid", () => {
+      return request(app)
+        .get("/api/articles?order=oldest&sort_by=whatcolumn&topic=mitch")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("sort_by 'whatcolumn' is not allowed.");
+        });
+    });
```
Comment (Geo-Beard) [inline __tests__/app.test.js:295]: You can add another test here for your greenlight list - check if order, sort_by, or topic is misspelled.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -175,6 +249,42 @@ describe("Error Handling", () => {
...
+  describe("Query Error Handling", () => {
+    it("GET- 400: Should respond with invalid query when passed an invalid column name in sort_by", () => {
+      return request(app)
+        .get("/api/articles?sort_by=whatcolumn")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("sort_by 'whatcolumn' is not allowed.");
+        });
+    });
+    it("GET- 400: Should respond with 400 when given an incorrect topic", () => {
```
Comment (Geo-Beard) [inline __tests__/app.test.js:270]: Should be 404 for non-existent topic

File: __tests__/app.test.js [hunk 5]
```diff
@@ -47,6 +47,80 @@ describe("/api/articles", () => {
         });
       });
   });
+  it("GET - 200: Should return only those with given topic.", () => {
```
Comment (Geo-Beard) [inline __tests__/app.test.js:50]: Add another test for a topic that has no articles - should return an empty array.

**Review by Geo-Beard — COMMENTED** [review]
A few comments to have a look at - nothing major - if you implement those changes feel free to merge and continue from there :)
